### PR TITLE
Mobile inference

### PR DESCRIPTION
**Pull request by playertr**
This PR addresses https://github.com/lojzezust/WaSR-T/issues/1 by adding and documenting a mobile-optimized version of WaSR-T.

The meat of the changes are in `mobile_wasr_t.py`, which uses Python inheritance to implement a few changes to the network without modifying the original `wasr_t.py` file. These changes are:
 - using the smaller lrasp_mobilenet_v3 backbone
 - reducing the latent channel dimension of several elements in the decoder
 - doing inference at 192x256 resolution

This PR also adds a script for real-time inference using gstreamer, a new `--mobile` flag for `train.py` and `predict_sequential.py`, and a new requirements.txt file specific to the Jetson Nano (which likely has different versions of PyTorch). Documentation for the new features is provided as an addendum to the README.

I'm open to any changes to this PR prior to merging -- in particular, I'd be willing to document the folder structure of the FolderDataset or to sequester all the new changes into a new folder if those would be positive changes.

### CODE REVIEW

**Comment by lojzezust** [comment]
Great work, @playertr! For completeness, I will also add the `--mobile` flag to other inference scripts.

Can you please provide the trained model weights for the new mobile model, so I can test it and include it in the release later?

**Comment by playertr** [comment]
Good idea. Here are the trained model weights:

https://drive.google.com/open?id=1zU_9Ut5movnX8pI4XTOvfEBSQbgMZdvA

They can be downloaded using gdown with

```
gdown --fuzzy -O lraspp_weights.ckpt https://drive.google.com/open?id=1zU_9Ut5movnX8pI4XTOvfEBSQbgMZdvA
```

**Comment by lojzezust** [comment]
Thanks! I've unified the `--mobile` and `--size` arguments across scripts. Can you verify that your `predict_gstreamer.py` script still works?

I will also run the inference on MODS to verify the accuracy of the mobile and the regular model.
Then after some minor documentation changes, I think we're good to go :)

**Comment by playertr** [comment]
Yes, I can confirm that the `predict_gstreamer.py` script still works with the mobile network on the Jetson Nano.

I was not able to confirm that the full WaSR-T network works with gstreamer because the Nano doesn't have enough VRAM, and my beefier desktop doesn't have a working gstreamer setup. (On x86, a working gstreamer setup requires building OpenCV from source to enable `cv2.CAP_GSTREAMER`, and has some confusing GTK graphics dependencies that appear to be broken in my conda environment.) I think it's okay not to test this because the gstreamer functionality is most relevant to Jetson Nano users.

I added the new arguments to the Mobile Inference documentation. If `--mobile` is omitted, the user gets a potentially confusing message about parameter size mismatch between the instantiated model and the loaded weights.

**Comment by lojzezust** [comment]
Awesome, thanks. I've added a table with mobile version results. The links to weights are placeholders and will be added once the PR is merged. 

I also retrained the mobile models myself to double-check the results. They performance gap is still quite large to the full model, but it should be good enough to get started. Interestingly, the model trained only on MaSTr1325 seems to perform better on MODS, compared to using additional out-of-distribution training examples. I think the smaller model is not able to generalize as well and thus the additional samples hurt the in-domain performance.

I've also updated the README slightly and moved the Jetson installation instructions to a separate file. Check if you're okay with the changes, and then we are ready to merge the PR.

**Comment by playertr** [comment]
Interesting! I like those documentation changes a lot.

Did you get similar results on MODS using the pretrained weights I provided versus the weights you trained? I believe I trained for like 500 epochs with a fairly low batch size, so the larger number of iterations may impact performance.

**Comment by lojzezust** [comment]
Using your weights, I get almost identical results as those I reported for MaSTr1478, so I'm guessing you trained using additional samples as well?

**Comment by playertr** [comment]
Yep, I used MaSTr1478 to train. It sounds like any additional differences in the training process didn't influence results on MODS. I think we're ready to merge.

**Comment by lojzezust** [comment]
Fusion-ha
